### ISSUE TITLE
Allow opening journals on mounted drives

### ISSUE THREAD

**Issue by jendrikseipp** [flathub/app.rednotebook.RedNotebook#8]
In issue https://github.com/jendrikseipp/rednotebook/issues/441 users report that only after using `flatpak override app.rednotebook.RedNoteBook --filesystem=/media` RedNotebook can open journals on mounted drives. @scx and @takluyver: is that option something that could be baked into the flatpak package?

**Comment by takluyver** [comment]
That specific location is platform dependent - on my (Fedora) system removable drives are mounted in `/run/media`. But it's technically possible to allow any filesystem access in the `finish-args` section of the manifest.

The bigger question is: should it? Flatpaks are designed to be sandboxed, and limiting filesystem access is an important part of that, so in some sense we're being a bad citizen if we give RedNotebook R/W access to the entire filesystem. We currently compromise by allowing access to the home directory, but nothing else.

There are two main ways to store data 'nicely' in the sandbox:

1. Storing data XDG directories like `$XDG_DATA_HOME` - Flatpak sets up data directories inside the sandbox for each app, so they can persist data. If you were writing RN from scratch for Flatpak, you'd probably start by storing the journal this way.
2. When you use GTK's open/save functions to prompt the user for a file, it should use a 'portal' to give the application temporary access to the file. But AFAIK, it is temporary: you can't reopen the file next time the app starts, unless the user grants access again.

Neither of these is a great fit for the model RedNotebook has, where you can select any directory for your journal, and it will remember it when you reopen the app. So maybe the only option is to give it more filesystem access. But I wanted to outline what the compromise is.

[Flatpak sandbox documentation](https://docs.flatpak.org/en/latest/sandbox-permissions.html#)

**Comment by takluyver** [comment]
See also:

- https://github.com/flatpak/flatpak/issues/2713
- https://github.com/flatpak/flatpak/issues/2185
- You can also use the [Flatseal](https://flathub.org/apps/details/com.github.tchx84.Flatseal) app to modify permissions without having to fiddle at the command line.

**Comment by jendrikseipp** [comment]
Thanks for the explanation! I believe adding `--filesystem=/media --filesystem=/run/media` is the most user-friendly option at the moment.

**Comment by takluyver** [comment]
Yup, I think I agree. If anything, the home directory access it already has is more dangerous.
